### PR TITLE
Add Java Solution to Bronze Why Did the Cows Cross the Road II

### PR DESCRIPTION
**Pull request by Navigator365**
_If any of the below doesn't apply to this Pull Request, mark the checkbox as
done._

i have returned
- [X] I have tested my code
- [X] I have followed the code style guidelines mentioned
      [here](https://usaco.guide/general/adding-solution/#code-conventions)
- [X] I have properly formatted the files according to the steps mentioned
      [here](https://usaco.guide/general/adding-solution#steps)

### CODE REVIEW

File: solutions/usaco-712.mdx [hunk 1]
```diff
@@ -112,6 +112,37 @@ int main() {
...
+		int ans = 0; 
+
+		// Iterate through all characters of the string
+		for (int a = 0; a < s.length(); a++)
+		for (int b = a + 1; b < s.length(); b++)
+			for (int c = b + 1; c < s.length(); c++)
+				for (int d = c + 1; d < s.length(); d++)
+					// If we find two entry/exit paths such that a < b < c < d, 
+					// the path of cows crossing (a, c) and (b, d) must intersect,
+					// so we have a crossing pair (only one-we can't count the pair twice)
```
Comment (SansPapyrus683) [inline solutions/usaco-712.mdx:134]:
```java
/*
 * multline comments should be
 * written like this

File: solutions/usaco-712.mdx [hunk 2]
```diff
@@ -112,6 +112,37 @@ int main() {
...
+	public static void main(String[] args) throws IOException {
+		Kattio io = new Kattio("circlecross");
+		String s = io.next();
+		int ans = 0; 
+
+		// Iterate through all characters of the string
+		for (int a = 0; a < s.length(); a++)
+		for (int b = a + 1; b < s.length(); b++)
+			for (int c = b + 1; c < s.length(); c++)
+				for (int d = c + 1; d < s.length(); d++)
```
Comment (SansPapyrus683) [inline solutions/usaco-712.mdx:131]: add braces

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

**Review by ryanchou-dev — APPROVED** [review]
rest looks good, just do the braces around the for loops

welcome back!

File: solutions/usaco-712.mdx (hunk 1 above)
Comment (Navigator365) [reply solutions/usaco-712.mdx]: you got it playa

**Review by SansPapyrus683 — APPROVED** [review]

**Pull request merged**